src/mlang/backend_compilers/bir_to_java.ml

```diff
@@ -0,0 +1,426 @@
+      Format.fprintf oc "%acalculationVariables[%d /*\"%a\"*/] = %s;@,"
+        (format_local_vars_defs var_indexes)
+        defs (get_var_pos var var_indexes) format_var_name var se
+  | TableVar (_, IndexTable es) ->
+      Format.fprintf oc "%a"
+        (fun fmt ->
+          IndexMap.iter (fun i v ->
+              let sv, defs = generate_java_expr v var_indexes in
+              Format.fprintf fmt "%acalculationVariables[%d /* %a */] = %s;@,"
+                (format_local_vars_defs var_indexes)
+                defs
+                (get_var_pos var var_indexes |> ( + ) i)
+                format_var_name var sv))
+        es
+      (* Format.fprintf oc "mCalculation.getTableVariables().put(\"%a\",Arrays.asList(%a));@,"
+         format_var_name var (Format.pp_print_list ~pp_sep:(fun oc _ -> Format.fprintf oc ", ") (fun
+         oc (_, var) -> let var, _ = generate_java_expr var var_indexes in Format.fprintf oc "%s"
+         var)) (IndexMap.bindings es) *)
+  | TableVar (_, IndexGeneric e) ->
+      Errors.raise_spanned_error "generic index table definitions not supported in the java backend"
+        (Pos.get_position e)
+  | InputVar -> assert false
+
+let generate_header (oc : Format.formatter) (class_name : string) : unit =
+  Format.fprintf oc
+    "@[// %s@,%s@,/**@,* Main class containing calculation logic@,*/@,public class %s {@]"
+    Prelude.message java_imports class_name
+
+let generate_input_handling (function_spec : Bir_interface.bir_function)
+    (var_indexes : variable_id VariableMap.t) (oc : Format.formatter) (split_threshold : int) =
+  let input_vars = List.map fst (VariableMap.bindings function_spec.func_variable_inputs) in
+  let print_header count =
+    Format.fprintf oc
+      "@[<hv 2>private static void loadInputVariables_%d(Map<String, MValue> inputVariables, \
+       MValue[] calculationVariables){@,"
+      count
+  in
+  let rec print_load_input curr oc len =
+    if curr <= len then (
+      Format.fprintf oc "loadInputVariables_%d(inputVariables, calculationVariables);@," curr;
+      print_load_input (curr + 1) oc len)
+  in
+  let format_input_var var count =
+    if count mod split_threshold = 0 then
+      print_header (if count > 0 then count / split_threshold else 0);
+    Format.fprintf oc
+      "calculationVariables[/*\"%a\"*/%d] = inputVariables.get(\"%s\") != null ? \
+       inputVariables.get(\"%s\") : MValue.mUndefined;@,"
+      format_var_name var (get_var_pos var var_indexes) (generate_name var) (generate_name var);
+    if (count + 1) mod split_threshold = 0 then Format.fprintf oc "@]@,@[}@]";
+    count + 1
+  in
+  let count = List.fold_left (fun count assign -> format_input_var assign count) 0 input_vars in
+  Format.fprintf oc
+    "}@,\
+     static void loadInputVariables(Map<String, MValue> inputVariables, MValue[] \
+     calculationVariables) {@,\
+     %a@,\
+     }"
+    (print_load_input 0) (count / split_threshold)
+
+let sanitize_str (s, p) =
```

Comment:
Maybe a little doc that says that the DGFiP source files are encoded with ISO something and that they might contain characters that Java doesn't like?

Reply:
Refactored into strings module for python and Java backends and given proper documentation